#! /bin/sh /usr/share/dpatch/dpatch-run
## dhclient.c.stale-pids.dpatch by  <apollock@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from Ubuntu via Martin Pitt to prevent multiple dhclients starting
## DP: up. (See #178885)

@DPATCH@
diff -urNad dhcp3-3.0.2/client/dhclient.c /tmp/dpep.lfPhxA/dhcp3-3.0.2/client/dhclient.c
--- dhcp3-3.0.2/client/dhclient.c	2005-06-15 12:59:10.433379868 +1000
+++ /tmp/dpep.lfPhxA/dhcp3-3.0.2/client/dhclient.c	2005-06-15 19:50:39.351829119 +1000
@@ -241,7 +241,7 @@
 	}
 
 	/* first kill of any currently running client */
-	if (release_mode) {
+	if (1) {
 		FILE *pidfd;
 		pid_t oldpid;
 		long temp;
@@ -251,11 +251,17 @@
 		if ((pidfd = fopen(path_dhclient_pid, "r")) != NULL) {
 			e = fscanf(pidfd, "%ld\n", &temp);
 			oldpid = (pid_t)temp;
+                        log_info ("There is already a pid file %s with pid %i", path_dhclient_pid, oldpid);
 
 			if (e != 0 && e != EOF) {
 				if (oldpid) {
-					if (kill(oldpid, SIGTERM) == 0)
-						unlink(path_dhclient_pid);
+					if (kill(oldpid, SIGTERM) == 0) {
+                                                log_info ("killed old client process, removed PID file");
+                                                unlink(path_dhclient_pid);
+                                        } else if (errno == ESRCH) {
+                                                log_info ("removed stale PID file");
+                                                unlink(path_dhclient_pid);
+                                        }
 				}
 			}
 			fclose(pidfd);
